main.go

```diff
@@ -74,28 +115,37 @@ func main() {
 		log.Println(err)
 	}
 
-	file, err := os.Open(filename)
-	if err != nil {
-		githubactions.Fatalf(fmt.Sprintf("opening file with filename: %v failed with error: %v", filename, err))
-	}
-
-	// decide name of file in GDrive
 	if name == "" {
+		file, err := os.Open(filename)
+		if err != nil {
+			githubactions.Fatalf(fmt.Sprintf("opening file with filename: %v failed with error: %v", filename, err))
+		}
 		name = file.Name()
 	}
 
-	f := &drive.File{
-		Name:    name,
-		Parents: []string{folderId},
+	if overwriteFlag {
+		r, err := svc.Files.List().Do()
```

Comment:
This is querying every single file in the directory, hence there is a bug in this diff. If there are files, but none of them match the `filename`, then nothing is uploaded to google drive. The expected behaviour would be to upload a new file.

Let's try querying the files by name instead using Q(). See https://pkg.go.dev/google.golang.org/api/drive/v3#FilesListCall.Q

Here is an explanation of the query syntax:
https://developers.google.com/drive/api/v3/search-files#query_string_examples

```
// Query for all files in google drive directory with name = <filename>
filenameQuery = fmt.Sprintf("name = '%s'", filename)
filesQueryCallResult, err := svc.Files.List().Q(filenameQuery).Do()

if len(filesQueryCallResult.Files == 0) {
  // upload file to drive
} else {
  // overwrite each file
}
```